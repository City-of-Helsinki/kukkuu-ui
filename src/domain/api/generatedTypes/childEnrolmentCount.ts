/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ChildEnrolmentCount
// ====================================================

export interface ChildEnrolmentCount_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many times a single user can participate events per year. Changing this will not affect any existing enrolments.
   */
  enrolmentLimit: number;
}

export interface ChildEnrolmentCount_child {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * How many past enrolments child has this year.
   */
  pastEnrolmentCount: number | null;
  project: ChildEnrolmentCount_child_project;
}

export interface ChildEnrolmentCount {
  child: ChildEnrolmentCount_child | null;
}

export interface ChildEnrolmentCountVariables {
  childId: string;
}
