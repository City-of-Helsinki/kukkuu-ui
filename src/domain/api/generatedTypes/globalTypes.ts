/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum EventParticipantsPerInvite {
  CHILD_AND_1_OR_2_GUARDIANS = "CHILD_AND_1_OR_2_GUARDIANS",
  CHILD_AND_GUARDIAN = "CHILD_AND_GUARDIAN",
  FAMILY = "FAMILY",
}

/**
 * An enumeration.
 */
export enum Language {
  EN = "EN",
  FI = "FI",
  SV = "SV",
}

export enum RelationshipTypeEnum {
  ADVOCATE = "ADVOCATE",
  OTHER_GUARDIAN = "OTHER_GUARDIAN",
  OTHER_RELATION = "OTHER_RELATION",
  PARENT = "PARENT",
}

export interface AddChildMutationInput {
  firstName?: string | null;
  lastName?: string | null;
  birthdate: any;
  postalCode: string;
  relationship?: RelationshipInput | null;
  languagesSpokenAtHome?: string[] | null;
  clientMutationId?: string | null;
}

export interface ChildInput {
  firstName?: string | null;
  lastName?: string | null;
  birthdate: any;
  postalCode: string;
  relationship?: RelationshipInput | null;
  languagesSpokenAtHome?: string[] | null;
}

export interface DeleteChildMutationInput {
  id: string;
  clientMutationId?: string | null;
}

export interface EnrolOccurrenceMutationInput {
  occurrenceId: string;
  childId: string;
  clientMutationId?: string | null;
}

export interface GuardianInput {
  firstName: string;
  lastName: string;
  phoneNumber?: string | null;
  language: Language;
  email?: string | null;
}

export interface RelationshipInput {
  type?: RelationshipTypeEnum | null;
}

export interface UnenrolOccurrenceMutationInput {
  occurrenceId: string;
  childId: string;
  clientMutationId?: string | null;
}

export interface UpdateChildMutationInput {
  id: string;
  firstName?: string | null;
  lastName?: string | null;
  birthdate?: any | null;
  postalCode?: string | null;
  relationship?: RelationshipInput | null;
  languagesSpokenAtHome?: string[] | null;
  clientMutationId?: string | null;
}

export interface UpdateMyProfileMutationInput {
  firstName?: string | null;
  lastName?: string | null;
  phoneNumber?: string | null;
  language?: Language | null;
  email?: string | null;
  clientMutationId?: string | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
