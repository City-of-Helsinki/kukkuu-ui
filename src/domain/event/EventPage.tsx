import { ReactElement } from 'react';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { IconArrowLeft } from 'hds-react';

import styles from './event.module.scss';
import PageWrapper from '../app/layout/PageWrapper';
import {
  EventQuery,
  ExternalTicketSystemEventQuery,
} from '../api/generatedTypes/graphql';
import { OccurrenceEvent } from './types/OccurrenceQueryTypes';

type EventProps = {
  event:
    | EventQuery['event']
    | OccurrenceEvent
    | ExternalTicketSystemEventQuery['event'];
  children?: ReactElement | Array<ReactElement | false>;
  success?: ReactElement;
  backTo?: string;
};

const EventPage = ({ event, children, success, backTo }: EventProps) => {
  const { t } = useTranslation();
  if (!event) return <></>;

  const backgroundImageStyle = event.image
    ? {
        backgroundImage: `url("${event.image}")`,
      }
    : {
        backgroundColor: 'var(--color-summer)',
      };

  return (
    <>
      <div
        className={styles.heroWrapper}
        style={backgroundImageStyle}
        title={event.imageAltText || ''}
      >
        {success}
        <div className={styles.backButtonWrapper}>
          <div className={styles.backButtonInnerWrapper}>
            {backTo && (
              <Link
                aria-label={t('common.backButton.label')}
                className={styles.backButton}
                to={backTo}
              >
                <IconArrowLeft className={styles.backButtonIcon} />
              </Link>
            )}
          </div>
        </div>
      </div>

      <PageWrapper className={styles.wrapper} title={event.name || ''}>
        <div className={styles.eventWrapper} role="main">
          <div className={styles.event}>
            <div className={styles.heading}>
              <h1>{event.name}</h1>
            </div>
            {children}
          </div>
        </div>
      </PageWrapper>
    </>
  );
};

export default EventPage;
