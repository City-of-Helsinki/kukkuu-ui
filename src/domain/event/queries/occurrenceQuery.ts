import gql from 'graphql-tag';

const occurrenceQuery = gql`
  query occurrenceQuery($id: ID!) {
    occurrence(id: $id) {
      id
      time
      remainingCapacity
      event {
        id
        image
        description
        shortDescription
        name
        duration
        participantsPerInvite
      }
      venue {
        id
        name
        address
        accessibilityInfo
        arrivalInstructions
        additionalInfo
        wwwUrl
        wcAndFacilities
      }
    }
  }
`;
export default occurrenceQuery;
