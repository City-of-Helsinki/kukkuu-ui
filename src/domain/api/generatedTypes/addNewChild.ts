/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AddChildMutationInput } from './globalTypes';

// ====================================================
// GraphQL mutation operation: addNewChild
// ====================================================

export interface addNewChild_addChild_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface addNewChild_addChild_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: addNewChild_addChild_child_project;
}

export interface addNewChild_addChild {
  child: addNewChild_addChild_child | null;
}

export interface addNewChild {
  /**
   * This mutation cannot be used before one has started using the service with "SubmitChildrenAndGuardianMutation".
   */
  addChild: addNewChild_addChild | null;
}

export interface addNewChildVariables {
  input: AddChildMutationInput;
}
