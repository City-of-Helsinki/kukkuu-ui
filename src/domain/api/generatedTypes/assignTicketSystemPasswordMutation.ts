/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssignTicketSystemPasswordMutationInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: assignTicketSystemPasswordMutation
// ====================================================

export interface assignTicketSystemPasswordMutation_assignTicketSystemPassword {
  /**
   * The assigned ticket system password
   */
  password: string | null;
}

export interface assignTicketSystemPasswordMutation {
  assignTicketSystemPassword: assignTicketSystemPasswordMutation_assignTicketSystemPassword | null;
}

export interface assignTicketSystemPasswordMutationVariables {
  input: AssignTicketSystemPasswordMutationInput;
}
