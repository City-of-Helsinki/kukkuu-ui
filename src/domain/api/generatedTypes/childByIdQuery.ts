/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { EventParticipantsPerInvite, RelationshipTypeEnum } from "./globalTypes";

// ====================================================
// GraphQL query operation: childByIdQuery
// ====================================================

export interface childByIdQuery_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface childByIdQuery_child_occurrences_edges_node_venue {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  address: string | null;
}

export interface childByIdQuery_child_occurrences_edges_node_event {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  /**
   * In minutes
   */
  duration: number | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  /**
   * An unique encoded reference id
   */
  referenceId: string | null;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_occurrences_edges_node_enrolments_edges_node | null;
}

export interface childByIdQuery_child_occurrences_edges_node_enrolments {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_occurrences_edges_node_enrolments_edges | null)[];
}

export interface childByIdQuery_child_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
  venue: childByIdQuery_child_occurrences_edges_node_venue;
  event: childByIdQuery_child_occurrences_edges_node_event;
  enrolments: childByIdQuery_child_occurrences_edges_node_enrolments;
}

export interface childByIdQuery_child_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_occurrences_edges_node | null;
}

export interface childByIdQuery_child_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_occurrences_edges | null)[];
}

export interface childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventNode {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
}

export interface childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventGroupNode {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
}

export type childByIdQuery_child_availableEventsAndEventGroups_edges_node = childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventNode | childByIdQuery_child_availableEventsAndEventGroups_edges_node_EventGroupNode;

export interface childByIdQuery_child_availableEventsAndEventGroups_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_availableEventsAndEventGroups_edges_node | null;
}

export interface childByIdQuery_child_availableEventsAndEventGroups {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_availableEventsAndEventGroups_edges | null)[];
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_pastEvents_edges_node_occurrences_edges_node | null;
}

export interface childByIdQuery_child_pastEvents_edges_node_occurrences {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_pastEvents_edges_node_occurrences_edges | null)[];
}

export interface childByIdQuery_child_pastEvents_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  shortDescription: string | null;
  image: string;
  imageAltText: string | null;
  participantsPerInvite: EventParticipantsPerInvite;
  occurrences: childByIdQuery_child_pastEvents_edges_node_occurrences;
}

export interface childByIdQuery_child_pastEvents_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_pastEvents_edges_node | null;
}

export interface childByIdQuery_child_pastEvents {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_pastEvents_edges | null)[];
}

export interface childByIdQuery_child_relationships_edges_node {
  /**
   * The ID of the object.
   */
  id: string;
  type: RelationshipTypeEnum | null;
}

export interface childByIdQuery_child_relationships_edges {
  /**
   * The item at the end of the edge
   */
  node: childByIdQuery_child_relationships_edges_node | null;
}

export interface childByIdQuery_child_relationships {
  /**
   * Contains the nodes in this connection.
   */
  edges: (childByIdQuery_child_relationships_edges | null)[];
}

export interface childByIdQuery_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: childByIdQuery_child_project;
  occurrences: childByIdQuery_child_occurrences;
  availableEventsAndEventGroups: childByIdQuery_child_availableEventsAndEventGroups | null;
  pastEvents: childByIdQuery_child_pastEvents | null;
  relationships: childByIdQuery_child_relationships;
}

export interface childByIdQuery {
  /**
   * The ID of the object
   */
  child: childByIdQuery_child | null;
}

export interface childByIdQueryVariables {
  id: string;
}
