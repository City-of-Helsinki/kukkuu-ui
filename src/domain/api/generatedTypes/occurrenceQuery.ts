/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { EventParticipantsPerInvite } from "./globalTypes";

// ====================================================
// GraphQL query operation: occurrenceQuery
// ====================================================

export interface occurrenceQuery_occurrence_event_eventGroup {
  /**
   * The ID of the object.
   */
  id: string;
}

export interface occurrenceQuery_occurrence_event {
  /**
   * The ID of the object.
   */
  id: string;
  image: string;
  imageAltText: string | null;
  description: string | null;
  shortDescription: string | null;
  name: string | null;
  /**
   * In minutes
   */
  duration: number | null;
  participantsPerInvite: EventParticipantsPerInvite;
  eventGroup: occurrenceQuery_occurrence_event_eventGroup | null;
}

export interface occurrenceQuery_occurrence_venue {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  address: string | null;
  accessibilityInfo: string | null;
  arrivalInstructions: string | null;
  additionalInfo: string | null;
  wwwUrl: string | null;
  wcAndFacilities: string | null;
}

export interface occurrenceQuery_occurrence {
  /**
   * The ID of the object.
   */
  id: string;
  time: any;
  remainingCapacity: number | null;
  event: occurrenceQuery_occurrence_event;
  venue: occurrenceQuery_occurrence_venue;
  childHasFreeSpotNotificationSubscription: boolean | null;
}

export interface occurrenceQuery {
  /**
   * The ID of the object
   */
  occurrence: occurrenceQuery_occurrence | null;
}

export interface occurrenceQueryVariables {
  id: string;
  childId?: string | null;
}
