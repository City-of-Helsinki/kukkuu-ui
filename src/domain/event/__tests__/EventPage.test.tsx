import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { occurrenceQuery_occurrence as OccurrenceQueryType } from '../../api/generatedTypes/occurrenceQuery';
import EventPage from '../EventPage';
import { EventParticipantsPerInvite } from '../../api/generatedTypes/globalTypes';

export const mockedOccurrenceNode: OccurrenceQueryType = {
  id: 'T2NjdXJyZW5jZU5vZGU6Mg==',
  time: '2020-03-08T04:00:00+00:00',
  remainingCapacity: 99,
  event: {
    id: 'zzaaz',
    name: 'event name',
    image: 'a',
    imageAltText: 'b',
    description: 'c',
    shortDescription: 'd',
    duration: 12,
    participantsPerInvite: EventParticipantsPerInvite.FAMILY,
    eventGroup: {
      id: 'e1',
    },
  },
  venue: {
    id: 'auppss',
    name: 'Musiikkitalo',
    address: '',
    accessibilityInfo: 'z',
    arrivalInstructions: 'y',
    additionalInfo: 'x',
    wwwUrl: 'https://example.com/z',
    wcAndFacilities: 'uio',
  },
  childHasFreeSpotNotificationSubscription: null,
};

it('renders snapshot correctly', () => {
  const element = shallow(
    <EventPage event={mockedOccurrenceNode.event} goBack={jest.fn()} />
  );
  expect(toJson(element)).toMatchSnapshot();
});
