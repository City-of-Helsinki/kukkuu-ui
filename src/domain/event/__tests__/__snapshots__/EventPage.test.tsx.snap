// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<Fragment>
  <div
    className="heroWrapper"
    style={
      Object {
        "backgroundImage": "url(\\"a\\")",
      }
    }
    title="b"
  >
    <div
      className="backButtonWrapper"
    >
      <div
        className="backButtonInnerWrapper"
      >
        <Button
          aria-label="Palaa"
          className="backButton"
          onClick={[Function]}
          variant="secondary"
        >
          <Icon
            className="backButtonIcon"
            src="arrowLeft.svg"
          />
        </Button>
      </div>
    </div>
  </div>
  <PageWrapper
    className="wrapper"
    title="event name"
  >
    <div
      className="eventWrapper"
      role="main"
    >
      <div
        className="event"
      >
        <div
          className="heading"
        >
          <h1>
            event name
          </h1>
        </div>
      </div>
    </div>
  </PageWrapper>
</Fragment>
`;
